lib/wicked/IPv6.pm

```diff
@@ -0,0 +1,133 @@
+# SUSE's openQA tests
+#
+# Copyright © 2020 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Helper module for IPv6 tests
+# Maintainer: Jose Lausuch <jalausuch@suse.com>
+
+package wicked::IPv6;
+
+use Mojo::Base 'wickedbase';
+use testapi;
+
+sub evaluate_step {
+    my ($self, $cmd, $value, $expected) = @_;
+    my $output = script_output($cmd);
+    record_info('output', "cmd: $cmd \noutput:\n$output");
+    die("$value is not expected") if ($expected == 0 && $output =~ $value);
+    die("$value is expected")     if ($expected == 1 && $output !~ $value);
+}
+
+sub setup_interface {
+    my ($self, %args) = @_;
+    die '$args{iface} is required'      unless $args{iface};
+    die '$args{ifcfg_type} is required' unless $args{ifcfg_type};
+    record_info("STEP", 'Setup ' . $args{ifcfg_type} . ' interface ' . $args{iface});
+    $self->get_from_data('wicked/' . $args{ifcfg_type} . '_address/ifcfg-eth0', '/etc/sysconfig/network/ifcfg-' . $args{iface});
+    $self->wicked_command('ifup', $args{iface});
+}
+
+sub wait_for_router_announcement {
+    my ($self, %args) = @_;
+    die '$args{iface} is required' unless $args{iface};
+    record_info("STEP", 'I let ' . $args{iface} . ' wait for a router announcement');
+    assert_script_run('rdisc6 ' . $args{iface});
+}
+
+sub retrigger_dhcp_requests {
+    my ($self, %args) = @_;
+    die '$args{iface} is required' unless $args{iface};
+    record_info("STEP", 'I retrigger DHCP requests');
+    $self->wicked_command('ifdown', $args{iface});
+    $self->wicked_command('ifup',   $args{iface});
+}
+
+sub obtain_address_by_dhcpv4 {
+    my ($self, %args) = @_;
+    die '$args{iface} is required' unless $args{iface};
+    my $expected = $args{should_not} == 1 ? 0 : 1;
```

Comment:
it is very confusing to get boolean variable just to invert it inside a function . Let's drop on of them or we getting `$args{expected}` directly as input param or we using `$args{should_not}` right away without inverting it

Reply:
ok, now I am passing `$args{should_not}` directly. Good suggestion, thanks.